### DISCUSSION TITLE
Getting DeZog working with ZEsarUX in VS Code

### DISCUSSION THREAD

**Discussion by DerekJC777** [maziac/DeZog#136] [Q&A] [answered]
I am having great difficulty getting DeZog to download compiled images to and debug within ZEsarUX. 

Here are my observations:
- It seems the instructions are out of date and no longer apply to the latest version of ZEsarUX (v11.0 released June 2024). The location of the remote protocol is no longer in "Settings"->"Debug"->"Remote protocol" to "Enabled"; I believe it's now in "Network"->"ZENG".
- I've tried using this and starting ZEsarUX with the command-line option "--enable-remoteprotocol", and neither seem to work.
- Currently, I get a list of Error: 'assert' errors at He.assert, Zr.parseSP, Zr.getRegValueByName, St.getRgValue, St.getSP, mn.getStackFromEmulator, and mn.getCallStack.
![Launching ZEsarUX errors](https://github.com/user-attachments/assets/bdd4eb64-c6da-494e-a595-3da0b3e7fa5f)

- My launch.json file contains this configuration:
 ```
       {
            "name": "ZEsarUX",
            "type": "dezog",
            "request": "launch",
            "remoteType": "zrcp",
            "zrcp": {
                "hostname": "localhost",
                "port": 10000
            },
            "topOfStack": "Stack_Top",
            "rootFolder": "${fileDirname}",
            "sjasmplus": [
                {
                    //"path": "${fileDirname}/${fileBasenameNoExtension}.lst",
                    "path": "${fileDirname}/${fileBasenameNoExtension}.sld",
                    "asm": "sjasmplus",
                    "useFiles": true,
                    "mainFile": "${fileDirname}/${fileBasenameNoExtension}.z80",
                    "srcDirs": [ "../lib" ]
                }
            ],
/*
            "disassemblerArgs": {
                "esxdosRst": true
            },
*/
            //"load": "${fileDirname}/${fileBasenameNoExtension}.sna",
            "load": "demo_colourbars.sna",
            //"skipInterrupt": false,
            "startAutomatically": true,
            "preLaunchTask": "sjasmplus"
        }
```

Any ideas about what is going on, what I'm missing or which setting is wrong?

**Comment by maziac** [comment, marked as answer]
Zesarux 11.0 is compatible with dezog.
Enabling of the zrcp protocol can still be done in the "Settings->Debug->Remote Protocol":
![Bildschirmfoto 2024-12-30 um 23 28 56](https://github.com/user-attachments/assets/fe73fbf6-54ed-4ed0-935c-1b2a2a8fe28e)

But it is harder to find now: you first have to enable the advanced menu:
![Bildschirmfoto 2024-12-30 um 23 31 11](https://github.com/user-attachments/assets/5a9fa09b-a4d6-419b-afa0-b6fe3808f3c5)

Anyhow it might be easier to use the commandline. Here is what I am using:
`zesarux --noconfigfile --machine tbblue --tbblue-fast-boot-mode --snap-no-change-machine --enable-esxdos-handler --enable-remoteprotocol --disable-autoframeskip --esxdos-root-dir .`

The errors you get "Currently, I get a list of Error: 'assert' errors at He.assert, Zr.parseSP, Zr.getRegValueByName, St.getRgValue, St.getSP, mn.getStackFromEmulator, and mn.getCallStack."
are probably because you connect to ZENG instead of zrcp. I haven't used ZENG yet, but I think it is used to remotely connect to another player, not for debugging.

There is an example project at: https://github.com/maziac/z80-sample-program

From your launch.json I can see that you are creating a sna file, i.e. for the ZX Spectrum. For Spectrum you can also use the internal simulator "zsim" instead of zesarux. 
The setup is much easier.
(Zesarux is required when you need more exact timing of the peripherals, e.g. the graphics, or if you need to emulator soundchip audio or if you need other machines like the ZX Next. Otherwise "zsim" can do as well.)

**Reply by DerekJC777** [reply]
I checked all the settings and they were all correct. The problem was the ZENG online server was enabled:

![ZEsarUX ZENG Online Server](https://github.com/user-attachments/assets/f1c20d1b-b7a6-47b0-b14c-b01b43d0ce68)

After disabling the ZENG server everything started working. The ZENG online server is found under the Network->ZENG Online Server menu.

**Comment by maziac** [comment]
Thanks.
I will add a note to the dezog description.
